Sources/TuistKit/Commands/TuistCommand.swift

```diff
@@ -44,19 +44,30 @@ public struct TuistCommand: ParsableCommand {
 
     public static func main(_ arguments: [String]? = nil) -> Never {
         let errorHandler = ErrorHandler()
-        var command: ParsableCommand
+        let executeCommand: () throws -> ()
         do {
             let processedArguments = Array(processArguments(arguments)?.dropFirst() ?? [])
-            if processedArguments.first == ScaffoldCommand.configuration.commandName {
-                try ScaffoldCommand.preprocess(processedArguments)
-            }
-            if processedArguments.first == InitCommand.configuration.commandName {
-                try InitCommand.preprocess(processedArguments)
-            }
-            if processedArguments.first == ExecCommand.configuration.commandName {
-                try ExecCommand.preprocess(processedArguments)
+            if Self.configuration.subcommands
+                .map({ $0._commandName })
+                .contains(processedArguments.first ?? "") {
+                if processedArguments.first == ScaffoldCommand.configuration.commandName {
+                    try ScaffoldCommand.preprocess(processedArguments)
+                }
+                if processedArguments.first == InitCommand.configuration.commandName {
+                    try InitCommand.preprocess(processedArguments)
+                }
+                if processedArguments.first == ExecCommand.configuration.commandName {
+                    try ExecCommand.preprocess(processedArguments)
+                }
+                let command = try parseAsRoot(processedArguments)
+                executeCommand = { try execute(command) }
+            } else {
+                executeCommand = {
+                    var customCommandArguments = processedArguments
+                    customCommandArguments[0] = "tuist-\(customCommandArguments[0])"
+                    try System.shared.runAndPrint(customCommandArguments)
+                }
```

Comment:
Should we check somehow if the command exists?
Otherwise if I do a typo (e.g. `tuist generata`), it might be hard to understand what's happening

Reply:
that is a very good point, yeah, we can check with `which xxx`, I'll do that.

Reply:
Yeah, that's a good idea. What about extracting that logic into its own component that we can test in isolation. Otherwise this `TuistCommand` will grow significantly in responsibilities.